server/datastore/mysql/teams.go

```diff
@@ -138,6 +144,20 @@ func loadUsersForTeamDB(ctx context.Context, q sqlx.QueryerContext, team *fleet.
 	}
 
 	team.Users = rows
+	team.UserCount = len(rows)
+
+	return nil
+}
+
+func loadHostCountForTeamDB(ctx context.Context, q sqlx.QueryerContext, team *fleet.Team) error {
+	stmt := `
+		SELECT COUNT(*) FROM hosts h
+		WHERE h.team_id = ?
+	`
+	if err := q.QueryRowxContext(ctx, stmt, team.ID).Scan(&team.HostCount); err != nil {
+		return ctxerr.Wrap(ctx, err, "load HostsCount for team")
+	}
+
```

Comment:
Presumably, as this host count information was missing, the frontend is doing an extra `GET /hosts/count?team_id=xyz` to get that info and display it? If so we should make sure that this extra API call will be removed so that we don't end up just adding latency doing unnecessary work. Just something worth mentioning on the ticket if it hasn't already been discussed.